golang/cosmos/x/swingset/keeper/extension_snapshotter.go

```diff
@@ -0,0 +1,270 @@
+package keeper
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/x/swingset/types"
+	vstoragetypes "github.com/Agoric/agoric-sdk/golang/cosmos/x/vstorage/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	snapshots "github.com/cosmos/cosmos-sdk/snapshots/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+var _ snapshots.ExtensionSnapshotter = &SwingsetSnapshotter{}
+
+// SnapshotFormat 1 is a proto message containing an artifact name, and the binary artifact data
+const SnapshotFormat = 1
+
+type activeExtensionSnapshot struct {
+	// The block height of the snapshot in progress
+	height uint64
+	// The logger for this snapshot
+	logger log.Logger
+	// The function to retrieve the exported artifacts
+	retrieveSwingStoreExport func() error
+	// The writer for extensions payloads
+	payloadWriter snapshots.ExtensionPayloadWriter
```

Comment:
```suggestion
	// height is the block height of this in-progress snapshot.
	height uint64
	// logger is the destination for this snapshot's log messages.
	logger log.Logger
	// retrieveSwingStoreExport fetches this snapshot's exported artifacts from ???.
	retrieveSwingStoreExport func() error
	// payloadWriter writes bytes into the under-construction protobuf message for a snapshot.
	// It may be called multiple times, and often is (e.g., once per artifact).
	payloadWriter snapshots.ExtensionPayloadWriter
```